src/task4feedback/simulator/analysis/plot.py

```diff
@@ -484,3 +487,126 @@ def make_data_plot(
         plt.tight_layout()
 
         plt.show()
+
+
+def make_dag_and_timeline(
+    simulator: SimulatedScheduler,
+    tasks: List[Mapping[TaskID, TaskInfo]],
+    plot_dag: bool = True,
+    plot_timeline: bool = True,
+    plot_data_movement: bool = False,
+):
+    recorders = simulator.recorders
+
+    try:
+        compute_task_record: Optional[ComputeTaskRecorder] = recorders.get(
+            ComputeTaskRecorder
+        )
+    except KeyError as e:  # raise error if not found
+        raise KeyError(e)
+
+    # Make color map of upto 32 colors in strings
+    colors = ["Red", "Green", "Blue", "Yellow", "Purple", "Orange", "Pink", "Brown"]
+    device_colors = {}
+
+    # Make dictionary of task results (Device it ran on, start time, end time)
+    task_results = {}
+    for taskid, task_record in compute_task_record.tasks.items():
+        task_result = {}
+        task_result["devices"] = task_record.devices
+        if task_record.devices[0] not in device_colors:
+            device_colors[task_record.devices[0]] = colors[len(device_colors)]
+        task_result["start_time"] = task_record.start_time
+        task_result["end_time"] = task_record.end_time
+        task_results[str(taskid)] = task_result
+
+    # Generate pydot graph from tasks
+    # Assign same color for same device
+    if plot_dag:
+        graph = pydot.Dot(graph_type="digraph")
+        for name, task_info in tasks.items():
+            name = str(name)
+            node = pydot.Node(
+                name=name,
+                style="filled",
+                fillcolor=device_colors[task_results[name]["devices"][0]],
+            )
+            graph.add_node(node)
+            for dep_id in task_info.dependencies:
+                dep_id = str(dep_id)
+                edge = pydot.Edge(dep_id, name)
+                graph.add_edge(edge)
+
+        graph.write_png("dag.png")
```

Comment:
later it would be great if you can input file names; so you can avoid overwritting files. also if you use scripts, you can generate multiple files; or you can change the file name from the scripts right after the execution completes.